src/PrestaShopBundle/Localization/DataSource/CLDR.php

```
@@ -0,0 +1,184 @@
+    {
+        $this->localeCode = (string)$localeCode;
+
+        return $this;
+    }
+
+    /**
+     * Get the data reader
+     *
+     * @return DataReaderInterface
+     * @throws Exception
+     */
+    public function getReader()
+    {
+        if (!isset($this->reader)) {
+            throw new Exception("Data reader has not been set");
+        }
+
+        return $this->reader;
+    }
+
+    /**
+     * Set the data reader
+     *
+     * @param DataReaderInterface $reader
+     *
+     * @return $this
+     */
+    public function setReader(DataReaderInterface $reader)
+    {
+        $this->reader = $reader;
+
+        return $this;
+    }
+
+    /**
+     * Get currency data by internal database identifier
+     *
+     * @param int $id
+     *
+     * @return array The currency data
+     * @throws InvalidArgumentException
+     */
+    public function getLocaleById($id)
+    {
+        if (!is_int($id)) {
+            throw new InvalidArgumentException('$id must be an integer');
+        }
+
+        return null;
+    }
+
+    /**
+     * Get locale data by code (either language code or IETF locale tag)
+     *
+     * @param string $code
+     *
+     * @return array The locale data
+     */
+    public function getLocaleByCode($code)
+    {
+        return $this->getReader()->getLocaleByCode($code);
+    }
+
+    /**
+     * Create a new locale in data source
+     *
+     * @param LocaleData $localeData
+     *
+     * @return int The id of newly created locale
+     */
+    public function createLocale(LocaleData $localeData)
+    {
+        // No write in CLDR data files
+        return $localeData->id;
+    }
+
+    /**
+     * Update an existing locale in data source
+     *
+     * @param LocaleData $localeData
+     *
+     * @return LocaleData The saved item
+     */
+    public function updateLocale(LocaleData $localeData)
+    {
+        // No write in CLDR data files
+        return $localeData;
```

It should throw an exception since it's not possible